scripts/test/Muon/utilities/muon_load_data_test.py

```diff
@@ -0,0 +1,229 @@
+    # TESTS
+    # ------------------------------------------------------------------------------------------------------------------
+
+    def test_load_data_initialized_as_empty(self):
+        data = MuonLoadData()
+        self.assert_empty(data)
+
+    def test_data_can_be_added_correctly_via_keyword_args(self):
+        data = MuonLoadData()
+        data.add_data(run=1234, workspace=[1], filename="C:\\dir1\\file1.nxs")
+
+        self.assertEqual(data.num_items(), 1)
+        self.assertEqual(data.contains_n(run=1234), 1)
+        self.assertEqual(data.contains_n(workspace=[1]), 1)
+        self.assertEqual(data.contains_n(filename="C:\\dir1\\file1.nxs"), 1)
+
+    def test_that_clear_empties_the_data_of_all_entries(self):
+        data = MuonLoadData()
+        data.add_data(run=1234, workspace=[1], filename="C:\\dir1\\file1.nxs")
+        data.add_data(run=1235, workspace=[2], filename="C:\\dir1\\file2.nxs")
+        data.add_data(run=1236, workspace=[3], filename="C:\\dir1\\file3.nxs")
+
+        data.clear()
+
+        self.assert_empty(data)
+
+    def test_that_adding_then_removing_single_item_leaves_data_empty(self):
+        data = MuonLoadData()
+
+        data.add_data(run=1234, workspace=[1], filename="C:\\dir1\\file1.nxs")
+        data.remove_data(run=1234)
+        self.assert_empty(data)
+
+        data.add_data(run=1234, workspace=[1], filename="C:\\dir1\\file1.nxs")
+        data.remove_data(workspace=[1])
+        self.assert_empty(data)
+
+        data.add_data(run=1234, workspace=[1], filename="C:\\dir1\\file1.nxs")
+        data.remove_data(filename="C:\\dir1\\file1.nxs")
+        self.assert_empty(data)
+
+    def test_that_remove_method_can_remove_several_entries_at_once(self):
+        data = MuonLoadData()
+
+        data.add_data(run=1234, workspace=[1], filename="C:\\dir1\\file1.nxs")
+        data.add_data(run=1234, workspace=[2], filename="C:\\dir1\\file2.nxs")
+        data.add_data(run=1234, workspace=[3], filename="C:\\dir1\\file3.nxs")
+
+        data.remove_data(run=1234)
+
+        self.assert_empty(data)
+
+    def test_that_remove_method_applies_OR_to_multiple_keyword_arguments(self):
+        data = MuonLoadData()
+
+        data.add_data(run=1234, workspace=[1], filename="C:\\dir1\\file1.nxs")
+        data.add_data(run=1235, workspace=[2], filename="C:\\dir1\\file2.nxs")
+        data.add_data(run=1236, workspace=[3], filename="C:\\dir1\\file3.nxs")
+
+        data.remove_data(run=1234, workspace=[2], filename="C:\\dir1\\file3.nxs")
+
+        self.assert_empty(data)
+
+    def test_that_contains_is_true_when_data_contains_entry_which_matches_to_a_single_keyword_and_false_otherwise(self):
+        data = MuonLoadData()
+
+        data.add_data(run=1234, workspace=[1], filename="C:\\dir1\\file1.nxs")
+        data.add_data(run=1235, workspace=[2], filename="C:\\dir1\\file2.nxs")
+        data.add_data(run=1234, workspace=[3], filename="C:\\dir1\\file3.nxs")
```

Comment:
the 3 above lines seem to be repeated a lot. Could we refactor them out to reduce repeated code?